lib/hooks.ts

```diff
@@ -47,10 +47,12 @@ export function useProductList(query?: QueryParams) {
     };
 }
 
-export function useProductInfo(pid: number, list: ListItem[]) {
+export function useProductInfo(pid: number, list?:ListItem[]) {
     const { context } = useSession();
     const params = new URLSearchParams({ context }).toString();
-    const product = list.find(item => item.id === pid);
+
+    let product: any; if(list.length > 0) { list = product }
```

Comment:
let's move the conditional to its own line. There should only be one variable/ statement per line. We should also avoid using `any` if we can. The inside of the brackets is also incorrect... we should copy the assignment we deleted on ln53 here...

 ```suggestion
    let product: ListItem[]; 
    
    if (list.length) { 
        product = list.find(item => item.id === pid);
    }
```